src/Control/Biegunka/Source/Git.hs

```diff
@@ -97,29 +94,43 @@ git_ :: URI -> FilePath -> Script 'Sources ()
 git_ u p = git' u p mempty
 
 updateGit :: URI -> FilePath -> Git -> IO (Maybe String)
-updateGit u p Git { _remote, _branch } =
+updateGit u p Git { _branch, _failIfAhead } =
   doesDirectoryExist p >>= \case
     True -> do
-      let rbr = _remote </> _branch
+      let rbr = "origin" </> _branch
       before <- gitHash
+      remotes <- lines `fmap` askGit p ["remote"]
+      when ("origin" `notElem` remotes) $
+        askGit' p ["remote", "add", "origin", u]
       -- Typical git: <https://stackoverflow.com/questions/11347712/git-fetch-only-for-current-branch>
-      _ <- askGit p ["fetch", _remote, _branch] -- TODO: track upstream
-      _ <- askGit p ["checkout", "-B", _branch, "--track", rbr] -- TODO: track upstream
-      _ <- askGit p ["merge", rbr, _branch] -- TODO: rebase
+      askGit' p ["fetch", "origin", _branch]
+      currentBranch <- (listToMaybe . lines) `fmap` askGit p ["rev-parse", "--abbrev-ref", "HEAD"]
+      when (currentBranch /= Just _branch) $
+        askGit' p ["checkout", "-B", _branch, "--track", rbr]
+      commitsAhead <- (not . null . lines) `fmap` askGit p ["rev-list", _branch ++ "..origin"]
+      if (commitsAhead && _failIfAhead)
```

Comment:
Should `git rebase` also happen when there are commits ahead and `_failIfAhead` is `False`?